swim/join_sender.go

```diff
@@ -334,6 +398,68 @@ func (j *joinSender) JoinCluster() ([]string, error) {
 	return nodesJoined, nil
 }
 
```

Comment:
What do you think of putting this logic in a method of an exponentialDelay object?